nep-11.mediawiki

```diff
@@ -0,0 +1,48 @@
+<pre>
+  NEP: 11
+  Title: Smart Contract Verification and Invocation with System Assets
+  Author: Alex DiCarlo (alex.dicarlo@neotracker.io)
+  Type: Standard
+  Status: Draft
+  Created: 2018-06-07
+  Replaces: NEP-7
+</pre>
+
+==Abstract==
+
+This NEP defines a scheme for smart contracts to handle system asset transfers.
+
+==Motivation==
+
+Smart contract authors cannot reliably refuse system asset transfers in smart contract invocations. They cannot refuse or react to system asset transfers made with transactions other than invocation transactions to the smart contract address. There is additional complexity in handling the possibility of multiple invocations in the same invocation transaction that expect system assets, typically requiring tracking between calls. At a high level, there are two main issues that this NEP aims to solve:
+
+# Allow smart contracts to refuse system asset transfers.
+# Allow smart contracts to react to system asset transfers.
```

Comment:
NEP-7 does exactly these.

Reply:
@RavenXce yes, this is a NEP to replace NEP-7 with a better design.